### ISSUE TITLE
How to reproduce the result stably?

### ISSUE THREAD

**Issue by wei-ln** [pyg-team/pytorch_geometric#2801]
## ❓ Questions & Help

Hi, 
    we have a problem with the reproduction question in PYG.  
    To reproduce the results of SAGE on Cora, we set a fixed seed (111 for example) in each repeat：    

        for ntimes in range(10):

            torch.backends.cudnn.deterministic = True
            torch.backends.cudnn.benchmark = False
            torch.backends.cudnn.enabled = False
            random.seed(111)
            torch.manual_seed(111)
            np.random.seed(111)
            torch.cuda.manual_seed_all(111)
    
            model, data = Net().to(device), data.to(device)
            optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)
            train() #train GNNs

However, we got different testaccuracy results:     

       [0.806, 0.808, 0.809, 0.806, 0.817, 0.8, 0.809, 0.808, 0.806, 0.808]

So, how to reproduce the result stably?

**Comment by rusty1s** [comment]
You can make use of our newly introduced `SparseTensor` class, see [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/sparse_tensor.html). I tried to explain this issue [here](https://github.com/rusty1s/pytorch_geometric/issues/2788#issuecomment-870502307).

**Comment by akihironitta** [comment]
Closing this in favor of https://github.com/pyg-team/pytorch_geometric/issues/2788#issuecomment-870502307.
